packages/flutter_tools/lib/src/base/features.dart

```diff
@@ -0,0 +1,245 @@
+  flutterWebFeature,
+  flutterLinuxDesktopFeature,
+  flutterMacOSDesktopFeature,
+  flutterWindowsDesktopFeature,
+];
+
+/// The [Feature] for flutter web.
+const Feature flutterWebFeature = Feature(
+  name: 'Flutter Web',
+  configSetting: 'flutter-web',
+  environmentOverride: 'FLUTTER_WEB',
+  master: FeatureSetting(
+    available: true,
+    enabledByDefault: false,
+  ),
+  dev: FeatureSetting(
+    available: true,
+    enabledByDefault: false,
+  ),
+);
+
+/// The [Feature] for macOS desktop.
+const Feature flutterMacOSDesktopFeature = Feature(
+  name: 'Flutter Desktop for macOS',
+  configSetting: 'flutter-macos-desktop',
+  environmentOverride: 'ENABLE_FLUTTER_DESKTOP',
+  master: FeatureSetting(
+    available: true,
+    enabledByDefault: false,
+  ),
+);
+
+/// The [Feature] for Linux desktop.
+const Feature flutterLinuxDesktopFeature = Feature(
+  name: 'Flutter Desktop for Linux',
+  configSetting: 'flutter-linux-desktop',
+  environmentOverride: 'ENABLE_FLUTTER_DESKTOP',
+  master: FeatureSetting(
+    available: true,
+    enabledByDefault: false,
+  ),
+);
+
+/// The [Feature] for Windows desktop.
+const Feature flutterWindowsDesktopFeature = Feature(
+  name: 'Flutter Desktop for Windows',
+  configSetting: 'flutter-windows-desktop',
+  environmentOverride: 'ENABLE_FLUTTER_DESKTOP',
+  master: FeatureSetting(
+    available: true,
+    enabledByDefault: false,
+  ),
+);
+
+/// A [FeatureFlags] that looks up values based on [Feature] definitions.
+class FlutterFeatureFlags implements FeatureFlags {
+  const FlutterFeatureFlags();
+
+  @override
+  bool get isLinuxEnabled => _isEnabled(flutterLinuxDesktopFeature);
+
+  @override
+  bool get isMacOSEnabled => _isEnabled(flutterMacOSDesktopFeature);
+
+  @override
+  bool get isWebEnabled => _isEnabled(flutterWebFeature);
+
+  @override
+  bool get isWindowsEnabled => _isEnabled(flutterWindowsDesktopFeature);
+
+  // Calculate whether a particular feature is enabled for the current channel.
+  static bool _isEnabled(Feature feature) {
+    final String currentChannel = FlutterVersion.instance.channel;
+    FeatureSetting featureSetting;
+    switch (currentChannel) {
+      case 'stable':
+        featureSetting = feature.stable;
+        break;
+      case 'beta':
+        featureSetting = feature.beta;
+        break;
+      case 'dev':
+        featureSetting = feature.dev;
+        break;
+      case 'master':
+      default:
+        featureSetting = feature.master;
+        break;
+    }
+    if (!featureSetting.available) {
+      return false;
+    }
+    bool isEnabled = featureSetting.enabledByDefault;
+    if (feature.configSetting != null) {
+      final bool configOverride = Config.instance.getValue(feature.configSetting);
+      if (configOverride != null) {
+        isEnabled = configOverride;
+      }
+    }
+    if (feature.environmentOverride != null) {
+      if (platform.environment[feature.environmentOverride] != null) {
```

Comment:
nit: would it make sense to check that the value is `'true'`?

Reply:
Done